server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CountersMetricsHandler.java

```diff
@@ -0,0 +1,90 @@
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.jboss.msc.service.ServiceController;
+
+/**
+ * Reflects runtime Counter attributes
+ *
+ * @author Vladimir Blagojevic
+ * @since 9.2
+ */
+public class CountersMetricsHandler extends AbstractRuntimeOnlyHandler {
+
+    public static final CountersMetricsHandler INSTANCE = new CountersMetricsHandler();
+
+    private static final int CACHE_CONTAINER_INDEX = 1;
+
+    @Override
+    protected void executeRuntimeStep(OperationContext context, ModelNode operation) throws OperationFailedException {
+        ModelNode attributeRead = operation.require(NAME);
+        final ModelNode result = new ModelNode();
+        final PathAddress address = PathAddress.pathAddress(operation.require(OP_ADDR));
+        final String cacheContainerName = address.getElement(CACHE_CONTAINER_INDEX).getValue();
+        final ServiceController<?> controller = context.getServiceRegistry(false)
+                .getService(CacheContainerServiceName.CACHE_CONTAINER.getServiceName(cacheContainerName));
+
+        String attribute = attributeRead.asString();
+        if (controller != null) {
+            DefaultCacheContainer cacheManager = (DefaultCacheContainer) controller.getValue();
+            CounterManagerConfiguration counterManagerConfiguration = extractCounterManagerConfiguration(cacheManager);
+
+            //we only have two attributes in CounterManagerConfiguration
+            if (CounterConfigurationMetrics.NUM_OF_OWNERS.toString().equals(attribute)) {
+                result.set(counterManagerConfiguration.numOwners());
+            } else if (CounterConfigurationMetrics.RELIABILITY.toString().equals(attribute)) {
+                result.set(counterManagerConfiguration.reliability().name());
+            }
+        }
+        context.getResult().set(result);
+    }
+
+    public void registerMetrics(ManagementResourceRegistration container) {
+        for (CounterConfigurationMetrics metric : CounterConfigurationMetrics.values()) {
+            container.registerMetric(metric.definition, this);
+        }
+    }
+
+    private CounterManagerConfiguration extractCounterManagerConfiguration(EmbeddedCacheManager cacheManager) {
+        GlobalComponentRegistry globalComponentRegistry = SecurityActions.getGlobalComponentRegistry(cacheManager);
+        CounterManagerConfiguration config = globalComponentRegistry.getGlobalConfiguration()
+                .module(CounterManagerConfiguration.class);
+        return config == null ? CounterManagerConfigurationBuilder.defaultConfiguration() : config;
+    }
+
+    public enum CounterConfigurationMetrics {
+
+        NUM_OF_OWNERS(MetricKeys.NUM_OF_OWNERS, ModelType.LONG),
```

Comment:
can't you reuse the attribute definition in `CacheContainerCountersResource`?

Reply:
I think I can't because these need to be set for runtime, the other are for non-runtime